code/core/inc/EnvironmentAndFrames.hpp

```diff
@@ -26,6 +27,12 @@ struct EnvironmentAndFrames
     double nu;
     double g;
     YamlRotation rot;
+
+    void set_rho_air(double value);
```

Comment:
Defining both a setter & a getter essentially makes the variable public: why do you need the setter?

Reply:
I need the setter to ensure the force models do not use rho_air directly when they want its value, because if it is unavailable it will raise an exception from boost::optional, and the user won't be able to understand that 'air rho' is missing in the input. By forcing the force models to get it with the getter, I ensure the right exception is thrown. The setter is needed because of the way SimulatorBuilder builds EnvironmentAndFrames.

Reply:
Basically, the member rho_air is public (like all other members), but its read-access is wrapped with the proper exception thrown if the value is not available.